lib/agent.js

```diff
@@ -174,12 +174,14 @@ Agent.prototype.start = function (opts) {
     return this
   } else if (this._conf.logLevel === 'trace') {
     var _ancestors = ancestors(module)
-    var basedir = path.dirname(process.argv[1])
     var stackObj = {}
     Error.captureStackTrace(stackObj)
 
+    // Attempt to load package.json from process.argv.
+    var pkg = null
     try {
-      var pkg = require(path.join(basedir, 'package.json'))
+      var basedir = path.dirname(process.argv[1] || '.')
+      pkg = require(path.join(basedir, 'package.json'))
```

Comment:
I see we're short circuiting to a value of `.` if `process.argv[1]` is not defined.